pkg/broker/handler/processors/filter/processor.go

```diff
@@ -93,7 +93,9 @@ func startSpan(ctx context.Context, trigger types.NamespacedName, event *event.E
 	return tracing.WithLogging(ctx, span), span
 }
 
-func (p *Processor) passFilter(ctx context.Context, attrs map[string]string, event *event.Event) bool {
+// PassFilter checks given event against attributes available in the attrs map to determine
+// if the event should pass or not.
+func PassFilter(ctx context.Context, attrs map[string]string, event *event.Event) bool {
```

Comment:
Making this an exported function seems a bit odd to me in terms of encapsulation and modularity. Don't take this as an objection though. We may want to consider a followup that refactors the ingress to use processors in the same style as fanout and retry.

Reply:
I do believe the code that decides if an event should be sent out by fan out, should be the exact same as the code that decides to drop events at ingress. Any discrepancy between those two methods would be problematic. So I opted to re-use the same method.

We could refactor into a common place where they both call a helper if that helps.